### PR TITLE
feat: support NebulaGraph DB, add NebulaGraphQAChain

### PR DESCRIPTION
**Pull request by wey-gu**
# feat: support NebulaGraph DB, add NebulaGraphQAChain

Bring the support of [NebulaGraph](http://github.com/vesoft-inc/nebula), the open-source distributed graph database with NebulaGraphQAChain

## Before submitting

- [x] a test for the integration - favor unit tests that do not rely on network access.
- [x] an example notebook showing its use

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

@tomasonjo
@vowelparrot

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -103,6 +103,7 @@ py-trello = {version = "^0.19.0", optional = true}
 momento = {version = "^1.5.0", optional = true}
 bibtexparser = {version = "^1.4.0", optional = true}
 pyspark = {version = "^3.4.0", optional = true}
+nebula3-python = {version = "^3.4.0", optional = true}
```
Comment (hwchase17) [inline pyproject.toml:106]: this needs to be in one of the dependdenyc groups

**Review by hwchase17 — COMMENTED** [review]
awesome. minor comment on pyproject, also happy to fix myself

**Comment by wey-gu** [comment]
Dear @hwchase17 thanks a lot for your change! I am still tuning the prompt locally and thus didn't push changes as you requested yet, sorry for this!

File: langchain/graphs/nebula_graph.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+        self.session_pool = self._get_session_pool()
+        self.schema = ""
+        # Set schema
+        try:
+            self.refresh_schema()
+        except Exception as e:
+            raise ValueError(f"Could not refresh schema. Error: {e}")
+
+    def _get_session_pool(self) -> SessionPool:
```
Comment (wey-gu) [inline langchain/graphs/nebula_graph.py:59]: Dear @hwchase17 I am getting error now, seems types still get resolved during runtime? I couldn't workaround this anyway, should we put back `-> Any` or?

```
---> 59 def _get_session_pool(self) -> SessionPool:
     60     assert all(
     61         [self.username, self.password, self.address, self.port, self.space]
     62     ), (
     63         "Please provide all of the following parameters: "
     64         "username, password, address, port, space"
     65     )
     67     from nebula3.Config import SessionPoolConfig

NameError: name 'SessionPool' is not defined
```

**Review by wey-gu — COMMENTED** [review]
@hwchase17 it seems with issues of `-> SessionPool` in runtime. Shall we move back to -> Any? I tried to put it with some hacks but failed.

File: langchain/graphs/nebula_graph.py (hunk 2 above)
Comment (wey-gu) [reply langchain/graphs/nebula_graph.py]: @hwchase17 now I reverted this typing part first to enable ut running/or it failed in end-to-end run, too.

**Comment by wey-gu** [comment]
Ready to merge now from my side @hwchase17 thanks! 🎉

**Comment by wey-gu** [comment]
Rebased to resolve the conflicts :)

**Comment by wey-gu** [comment]
Thanks a lot, @hwchase17 for the quick support and review!

**Pull request merged**